docs/custom_node_development.md

```diff
@@ -0,0 +1,126 @@
+# Custom Node Development Guide
+
+## Overview
+
+Custom node in OpenVINO Model Server simplifies linking deep learning models into a complete pipelines even if the inputs and output
+of the sequential models do not fit. In many cases, output of one model can not be directly passed to another one.
+The data might need to be analysed, filtered or converted to difference format. Those operations can not be easily implemented
+in AI frameworks or are just not supported. Custom node address this challenge. They allows employing a dynamic library
+developed in C++ or C to perform arbitrary data transformations. 
+
+## Custom Node API
+
+The custom node library must implemented the API interface defined in [custom_node_interface.h](../src/custom_node_interface.h).
+The interface is defined in `C` to simplify compatibility with various compilers. The library could use third party components
+linked statically or dynamically. OpenCV is a built in component in OVMS which could be used to perform manipulation on the image
+data. 
+
+Below are explained the data structures and functions defined in the API header. 
+
+### "CustomNodeTensor" struct 
+
+The CustomNodeTensor struct consist of several fields defining the data in the output and input of the node execution.
+Because it has constant length, it can carry multiple inputs. Custom node can generate results based on multiple inputs 
```

Comment:
What do you mean with "Because it has constant length, it can carry multiple inputs."?